applications/GeoMechanicsApplication/custom_strategies/schemes/geomechanics_time_integration_scheme.hpp

```diff
@@ -0,0 +1,360 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Richard Faasse
+//
+#pragma once
+
+#include "solving_strategies/schemes/scheme.h"
+
+namespace Kratos {
+
+template <class TSparseSpace, class TDenseSpace>
+class GeoMechanicsTimeIntegrationScheme : public Scheme<TSparseSpace, TDenseSpace> {
+public:
+    using BaseType = Scheme<TSparseSpace, TDenseSpace>;
+    using DofsArrayType = typename BaseType::DofsArrayType;
+    using TSystemMatrixType = typename BaseType::TSystemMatrixType;
+    using TSystemVectorType = typename BaseType::TSystemVectorType;
+    using LocalSystemVectorType = typename BaseType::LocalSystemVectorType;
+    using LocalSystemMatrixType = typename BaseType::LocalSystemMatrixType;
+
+    void GetDofList(const Element& rElement,
+                    Element::DofsVectorType& rDofList,
+                    const ProcessInfo& rCurrentProcessInfo) override
+    {
+        if (IsActive(rElement))
+            rElement.GetDofList(rDofList, rCurrentProcessInfo);
+    }
+
+    void GetDofList(const Condition& rCondition,
+                    Condition::DofsVectorType& rDofList,
+                    const ProcessInfo& rCurrentProcessInfo) override
+    {
+        if (IsActive(rCondition))
+            rCondition.GetDofList(rDofList, rCurrentProcessInfo);
+    }
+
+    void EquationId(const Element& rElement,
+                    Element::EquationIdVectorType& rEquationId,
+                    const ProcessInfo& rCurrentProcessInfo) override
+    {
+        if (IsActive(rElement))
+            rElement.EquationIdVector(rEquationId, rCurrentProcessInfo);
+    }
+
+    void EquationId(const Condition& rCondition,
+                    Condition::EquationIdVectorType& rEquationId,
+                    const ProcessInfo& rCurrentProcessInfo) override
+    {
+        if (IsActive(rCondition))
+            rCondition.EquationIdVector(rEquationId, rCurrentProcessInfo);
+    }
+
+    void Initialize(ModelPart& rModelPart) override
+    {
+        KRATOS_TRY
+
+        SetTimeFactors(rModelPart);
+
+        Scheme<TSparseSpace, TDenseSpace>::mSchemeIsInitialized = true;
+
+        KRATOS_CATCH("")
+    }
+
+    void InitializeSolutionStep(ModelPart& rModelPart,
+                                TSystemMatrixType& A,
+                                TSystemVectorType& Dx,
+                                TSystemVectorType& b) override
```

Comment:
Since these parameters are not used in the function's body, I would remove their names (to avoid messages about unused parameters).
```suggestion
                                TSystemMatrixType&,
                                TSystemVectorType&,
                                TSystemVectorType&) override
```

Reply:
Done!